core/arch/arm/kernel/secure_partition.c

```diff
@@ -603,6 +603,71 @@ static TEE_Result handle_fdt_dev_regions(struct sp_ctx *ctx, void *fdt)
 	return TEE_SUCCESS;
 }
 
+static TEE_Result swap_sp_endpoints(uint32_t endpoint_id,
+				    uint32_t new_endpoint_id)
+{
+	struct sp_session *session = sp_get_session(endpoint_id);
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t manifest_endpoint_id = 0;
+
+	/*
+	 * We don't know in which order the SPs are loaded. It can
+	 * be that the endpoint id defined in the manifest could already
+	 * be generated by new_session_id() and used by another SP. If
+	 * this is the case, we swap the ID's of the two SPs.
+	 * We also have to make sure that the ID's are not defined twice in the
+	 * manifest.
+	 */
+
+	/* The endpoint id was not assigned yet */
+	if (!session)
+		return TEE_SUCCESS;
+
+	/*
+	 * Read the manifest file from the SP who originally had the endpoint.
+	 * We can safely swap the endpoint ID's if the manifest file doesn't
+	 * have an endpoint ID defined.
+	 */
+	if (!sp_dt_get_u32(session->fdt, 0, "id", &manifest_endpoint_id)) {
+		if (manifest_endpoint_id == endpoint_id) {
```

Comment:
If "id" is found in the manifest, can it be anything but what we already have in `session->endpoint_id`?

Reply:
You are right. It's not possible, dropped the extra check.

Reply:
`assert(manifest_endpoint_id == endpoint_id);` as extra check?

Reply:
Ack